controllers/template/authconfig.yaml

```diff
@@ -1,33 +1,74 @@
 apiVersion: authorino.kuadrant.io/v1beta1
 kind: AuthConfig
 metadata:
-  name: odh-dashboard-protection
+  name: $ODH_NS-protection
   namespace: $ODH_NS
   labels:
     authorino/topic: odh
 spec:
   hosts:
-    - "${ODH_ROUTE}"
+  - "${ODH_ROUTE}"
   identity:
-    - name: kubernetes-users
-      kubernetes:
-        audiences:
-          - "https://kubernetes.default.svc"
+  - name: authorized-service-accounts
+    kubernetes:
+      audiences:
+      - $ODH_NS-api
+  - name: anonymous-grpc
+    priority: 1
+    anonymous: {}
+    when:
+    - selector: context.request.http.headers.mm-vmodel-id
+      operator: matches
+      value: .+
+    extendedProperties:
+    - name: modelid
+      valueFrom:
+        authJSON: context.request.http.headers.mm-vmodel-id
+  - name: anonymous-http
+    priority: 1
+    anonymous: {}
+    when:
+    - selector: context.request.http.headers.mm-vmodel-id
+      operator: matches
+      value: ^$
+    extendedProperties:
+    - name: modelid
+      valueFrom:
+        authJSON: context.request.http.path.@extract:{"sep":"/","pos":2}
+  metadata:
+  - name: access
+    http:
+      endpoint: http://odh-model-controller-metrics-service.opendatahub.svc.cluster.local:8080/model/?ns={context.request.http.host.@extract:{"sep":".","pos":0}}&modelid={auth.identity.modelid}
+    #cache:
+    #  key:
+    #    valueFrom:
+    #      authJSON: {context.request.http.host.@extract:{"sep":".","pos":0}}-{auth.identity.modelid}
+    #  ttl: 300
+    when:
+    - selector: auth.identity.anonymous
+      operator: eq
+      value: "true"
   authorization:
-    - name: k8s-rbac
-      kubernetes:
-        user:
-          valueFrom: { authJSON: auth.identity.username }
-  response:
-    - name: x-auth-data
+    - name: anonymous
+      when:
+      - selector: auth.identity.anonymous
+        operator: eq
+        value: "true"
       json:
-        properties:
-          - name: username
-            valueFrom: { authJSON: auth.identity.username }
+        rules:
+        - selector: auth.metadata.access.anonymous
+          operator: eq
+          value: "true"
   denyWith:
     unauthenticated:
-      message:
-        value: "Access denied"
+      code: 302
+      headers:
+      - name: Location
+        valueFrom:
+          authJSON: http://some-service?redirect_to={context.request.http.path}
     unauthorized:
-      message:
-        value: "Unauthorized"
+      code: 403
+      headers:
+      - name: Location
+        valueFrom:
+          authJSON: http://some-service?redirect_to={context.request.http.path}
```

Comment:
Are we planning to change those `authJSON` values? How and from where should we take those?

Reply:
This is a bit of an open question. I think in the current form these are more "examples" then real values as they are dummy and the chosen auth form does not support a kind of "start flow" type thing. 

This is now a User controlled AuthConfig. If they use some external auth server, they _could_ choose to auto redirect there if the request is not auth to e.g. to start the oauth flow. 

But maybe these should be commented out or moved to docs.